### PR TITLE
Adding gempy_engine

### PR DESCRIPTION
**Pull request by SophieCurinier**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by SophieCurinier** [comment]
Hey @conda-forge/help-python ,

my recipe is ready! Thank you for your help

File: recipes/gempy_engine/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{% set name = "gempy_engine" %}
```
Comment (synapticarbors) [inline recipes/gempy_engine/meta.yaml:1]: Let's make this `gempy-engine` to match how it's installed via `pip`.

File: recipes/gempy_engine/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+{% set version = "2024.2.0" %}
+{% set python_min = "3.10" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
```
Comment (synapticarbors) [inline recipes/gempy_engine/meta.yaml:10]:
```suggestion
  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/gempy_engine-{{ version }}.tar.gz
```

File: recipes/gempy_engine/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  home: https://github.com/gempy-project/gempy_engine
+  summary: A Python package necessary for GemPy.
+  description: |
+    A Python package necessary for GemPy.
+
+    This package is not maintained by its author.
+  license: EUPL-1.2
+  license_file:
+    - LICENSE
+    - AUTHORS.rst
```
Comment (synapticarbors) [inline recipes/gempy_engine/meta.yaml:47]: Is this a license file?

File: recipes/gempy_engine/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 35bd3e764305fd49955713703913e4619e12a7780439f342c0616c2c361e335a
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  host:
```
Comment (synapticarbors) [inline recipes/gempy_engine/meta.yaml:19]: Does this package use `setuptools_scm`?

File: recipes/gempy_engine/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    - python {{ python_min }}
+    - pip
+
+about:
+  home: https://github.com/gempy-project/gempy_engine
+  summary: A Python package necessary for GemPy.
+  description: |
+    A Python package necessary for GemPy.
+
+    This package is not maintained by its author.
```
Comment (synapticarbors) [inline recipes/gempy_engine/meta.yaml:43]: No need to note this

File: recipes/gempy_engine/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  number: 0
+
+requirements:
+  host:
+    - python {{ python_min }}
+    - pip
+    - setuptools
+  run:
+    - python >={{ python_min }}
+    - numpy >=1.23
```
Comment (synapticarbors) [inline recipes/gempy_engine/meta.yaml:25]: Upstream only says `numpy`. Is there a reason to constrain the version?

**Review by synapticarbors — CHANGES_REQUESTED** [review]

File: recipes/gempy_engine/meta.yaml (hunk 3 above)
Comment (SophieCurinier) [reply recipes/gempy_engine/meta.yaml:47]: Not really, it was the author file

File: recipes/gempy_engine/meta.yaml (hunk 6 above)
Comment (SophieCurinier) [reply recipes/gempy_engine/meta.yaml]: Not really, i corrected it

File: recipes/gempy_engine/meta.yaml (hunk 5 above)
Comment (SophieCurinier) [reply recipes/gempy_engine/meta.yaml]: Thx. Removed

File: recipes/gempy_engine/meta.yaml (hunk 4 above)
Comment (SophieCurinier) [reply recipes/gempy_engine/meta.yaml:19]: Indeed, added it

File: recipes/gempy_engine/meta.yaml (hunk 2 above)
Comment (SophieCurinier) [reply recipes/gempy_engine/meta.yaml]: Thx. Corrected

File: recipes/gempy_engine/meta.yaml (hunk 1 above)
Comment (SophieCurinier) [reply recipes/gempy_engine/meta.yaml]: Corrected

**Comment by SophieCurinier** [comment]
Thank you @synapticarbors! Everything reviews have been done

**Pull request merged**